docs/Contributing/high-level-architecture.md

```diff
@@ -0,0 +1,192 @@
+# High level architecture
+
+- [Overview](#overview)
+- [Main System Components](#main-system-components)
+
+## Overview
+
+Add text
+
+## Main System Components
+
+```mermaid
+graph LR;
+    
+    subgraph Development
+        fleet_release_owner[Fleet Release<br>Owner];
+    end
+
+    subgraph Agent
+        orbit[orbit];
+        desktop[Fleet Desktop<br>Tray App];
+        osqueryd[osqueryd];
+        desktop_browser[Fleet Desktop<br> from Browser];
+    end
+
+    subgraph Customer Cloud
+        fleet_server[Fleet<br>Server];
+        db[DB];
+        redis[Redis<br>Live queries' results <br>go here];
+        prometheus[Prometheus Server];
+    end
+
+    subgraph FleetDM Cloud
+        tuf["<a href=https://theupdateframework.io/>TUF</a> file server<br>(default: <a href=https://tuf.fleetctl.com>tuf.fleetctl.com</a>)"];
+        datadog[DataDog metrics]
+        heroku[Usage Analytics<br>Heroku]
+        log[Send logs to optional<br> external location]
+    end
+
+    subgraph Customer Admin
+        frontend[frontend code]
+    end
+
+
+    fleet_release_owner -- "Release Process" --> tuf;
+
+    orbit -- "Fleet Orbit API (TLS)" --> fleet_server;
+    orbit -- "Auto Update (TLS)" --> tuf;
+    desktop -- "Fleet Desktop API (TLS)" --> fleet_server;
+    osqueryd -- "osquery<br>remote API (TLS)" --> fleet_server;
+    desktop_browser -- "My Device API (TLS)" --> fleet_server;
+
+    heroku -- "Metrics from all customers" --> datadog;
+
+    fleet_server <== "Read/Write" ==> db;
+    fleet_server <== "Read/Write" ==> redis;
+    redis <==> db;
+
+    prometheus ==> fleet_server;
+    fleet_server -- "metrics" --> heroku;
+    fleet_server -- "queries results" --> log;
+
+    frontend <== "API" ==> fleet_server;
+
+```
+
```

Comment:
`frontend` is a bit generic. We have Web GUI, REST API, and `fleetctl`

Reply:
Fixed